atlas-2024-odfr/mkjson.py

```diff
@@ -0,0 +1,210 @@
+                my_files = []
+                for afile in my_files_dict:
+                    my_files += [ {'filename':afile,
+                                   'checuksum':my_files_dict[afile]['checksum'],
+                                   'size':my_files_dict[afile]['size'],
+                                   'events':my_files_dict[afile]['events'],
+                                   'type':my_files_dict[afile]['type'],
+                                   'uri_root':my_files_dict[afile]['uri'] } ]
+                # Final check that we have at least one file
+                if len(my_files)==0:
+                    print(f'No files identified for {dataset_line.strip()}')
+                    continue
+            else:
+                # This is a list of metadata for the MC samples
+                filename = 'MC_'+dataset_line.split()[2].strip().replace('\\','')+'_filelist.json'
+                # Get the list of files
+                my_did = dataset_line.split()[0].strip()
+                my_full_did_names = [ x for x in mc_json['file_locations'] if '.'+my_did+'.' in x ]
+                if len(my_full_did_names)>1:
+                    print(f'Warning: Found multiple matching DIDs from {my_did} : {my_full_did_names}')
+                elif len(my_full_did_names)==0:
+                    print(f'Found no matching DIDs from {my_did} : {my_full_did_names}')
+                    continue
+                my_full_did_name = my_full_did_names[0]
+                # Grab the dictionary of file metadata from the json file
+                my_files_dict = mc_json['file_locations'][my_full_did_name]
+                # Convert the dictionary into the format we want for the open data portal
+                my_files = []
+                for afile in my_files_dict:
+                    my_files += [ {'filename':afile,
+                                   'checuksum':my_files_dict[afile]['checksum'],
+                                   'size':my_files_dict[afile]['size'],
+                                   'events':my_files_dict[afile]['events'],
+                                   'type':my_files_dict[afile]['type'],
+                                   'uri_root':my_files_dict[afile]['uri'] } ]
+                # Final check that we have at least one file
+                if len(my_files)==0:
+                    print(f'No files identified for {my_full_did_name} from {my_did}')
+                    continue
+
+            # Set the metadata in the `super` (open data portal record) json
+            my_json['files'] += [ { 'filename':filename } ]
+            # Now open that file and write the file names there
+            with open(output_directory+'/'+filename,'w') as dataset_filelist_file:
+                json.dump( my_files , dataset_filelist_file )
+
+    # Write myself a json file
+    with open(output_directory+'/'+adataset.replace('.txt','.json'),'w') as outfile:
+        json.dump( my_json , outfile )
+
```

Comment:
Also, the `mkjson.py` script produces the following output files:

```
test/mc_boson_nominal.json
test/mc_exotics_nominal.json
test/mc_higgs_nominal.json
test/mc_higgs_systematics.json
test/mc_jet_nominal.json
test/mc_jet_systematics.json
test/mc_susy_nominal.json
test/mc_top_nominal.json
test/mc_top_systematics.json
test/pp_2015_data_p6026_tids.json
test/pp_2016_data_p6026_tids.json
```

If we would like them to be directly usable in open data repository as fixtures, it would be good to name the files differently. At the very least use leading `atlas-2024-...` prefix so that they would be nicely findable in https://github.com/cernopendata/opendata.cern.ch/tree/master/cernopendata/modules/fixtures/data/records

Reply:
Changed the names to be a little nicer, include the suggested prefix, use "-", and be configurable. Further suggestions welcome of course.